File: generar_dataset.py
import csv
import random

vehiculos = ["Moto", "Bicicleta", "Camioneta"]

with open("entregas_camba.csv", "w", newline="", encoding="utf-8") as f:
    writer = csv.writer(f)
    writer.writerow(["distancia_km", "tiempo_osrm_min", "vehiculo", "hora", "dia_semana", "lluvia", "trafico", "zonas_bloqueadas", "tiempo_real_min"])
    
    for _ in range(500):
        vehiculo = random.choice(vehiculos)
        distancia = round(random.uniform(1.0, 15.0), 1)
        
        # Base OSRM time
        if vehiculo == "Bicicleta":
            tiempo_osrm = int(distancia * 4) 
        elif vehiculo == "Moto":
            tiempo_osrm = int(distancia * 2.5)
        else:
            tiempo_osrm = int(distancia * 3)
            
        hora = random.randint(8, 22)
        dia = random.randint(0, 6)
        lluvia = 1 if random.random() < 0.2 else 0
        trafico = 1 if (17 <= hora <= 19 or 7 <= hora <= 9) or random.random() < 0.3 else 0
        zonas = random.randint(0, 3) if random.random() < 0.4 else 0
        
        tiempo_real = tiempo_osrm
        
        if trafico:
            tiempo_real += random.randint(5, 15)
        if lluvia:
            tiempo_real += random.randint(4, 12)
        if zonas > 0:
            tiempo_real += zonas * random.randint(2, 5)
            
        # Add some random noise
        tiempo_real += random.randint(-2, 3)
        if tiempo_real < tiempo_osrm:
            tiempo_real = tiempo_osrm + random.randint(0, 2)
            
        writer.writerow([distancia, tiempo_osrm, vehiculo, hora, dia, lluvia, trafico, zonas, tiempo_real])

File: generar_dataset.js
const fs = require('fs');

const vehiculos = ["Moto", "Bicicleta", "Camioneta"];
let csvContent = "distancia_km,tiempo_osrm_min,vehiculo,hora,dia_semana,lluvia,trafico,zonas_bloqueadas,tiempo_real_min\n";

for (let i = 0; i < 500; i++) {
    const vehiculo = vehiculos[Math.floor(Math.random() * vehiculos.length)];
    const distancia = parseFloat((Math.random() * 14 + 1).toFixed(1));
    
    let tiempo_osrm = 0;
    if (vehiculo === "Bicicleta") tiempo_osrm = Math.floor(distancia * 4);
    else if (vehiculo === "Moto") tiempo_osrm = Math.floor(distancia * 2.5);
    else tiempo_osrm = Math.floor(distancia * 3);
    
    const hora = Math.floor(Math.random() * 15) + 8; // 8 to 22
    const dia = Math.floor(Math.random() * 7); // 0 to 6
    const lluvia = Math.random() < 0.2 ? 1 : 0;
    const trafico = ((hora >= 17 && hora <= 19) || (hora >= 7 && hora <= 9) || Math.random() < 0.3) ? 1 : 0;
    const zonas = Math.random() < 0.4 ? Math.floor(Math.random() * 4) : 0;
    
    let tiempo_real = tiempo_osrm;
    if (trafico) tiempo_real += Math.floor(Math.random() * 11) + 5;
    if (lluvia) tiempo_real += Math.floor(Math.random() * 9) + 4;
    if (zonas > 0) tiempo_real += zonas * (Math.floor(Math.random() * 4) + 2);
    
    tiempo_real += Math.floor(Math.random() * 6) - 2;
    if (tiempo_real < tiempo_osrm) tiempo_real = tiempo_osrm + Math.floor(Math.random() * 3);
    
    csvContent += `${distancia},${tiempo_osrm},${vehiculo},${hora},${dia},${lluvia},${trafico},${zonas},${tiempo_real}\n`;
}

fs.writeFileSync('entregas_camba.csv', csvContent, 'utf-8');
console.log("entregas_camba.csv generado con éxito.");
